tensorflow/python/ops/variables.py

```diff
@@ -226,7 +226,12 @@ def __init__(self,
           expected_shape=expected_shape)
 
   def __str__(self):
-    return str(self._snapshot)
+    return "<Variable name='%s' shape=%s dtype=%s>" % (
+            self.name(), self.get_shape(), self.dtype().name)
```

Comment:
For consistency with the repr for `tf.Tensor`, can you please change the format string to:

`<tf.Variable '%s' shape=%s dtype=%s>`

Also, I think it should be `self.dtype` rather than `self.dtype()`.